tests/Unit/FeedXMLGenerationTest.php

```diff
@@ -0,0 +1,405 @@
+		$xml     = $sale_price_method( $product );
+		$this->assertEquals( "", $xml );
+
+		// Sale price set.
+		$product = \WC_Helper_Product::create_simple_product(
+			true,
+			array(
+				"regular_price" => 15,
+				"sale_price"    => 5
+			)
+		);
+		$xml     = $sale_price_method( $product );
+		$this->assertEquals( "<sale_price>5USD</sale_price>", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertySalePriceVariableProductXML() {
+		$price_method      = $this->getProductsXmlFeedAttributeMethod( 'g:price' );
+		$product           = new \WC_Product_Variable();
+		$variation_product = \WC_Helper_Product::create_variation_product( $product );
+		/* In UT flow we need to fetch the product again from the DB after creation.
+		 * This ensures correct initialization of visible variations.
+		 * Without that the variable price methods think that we don't have visible children.
+		 * Quirk of create_variation_product.
+		 */
+		$product = wc_get_product( $variation_product->get_id() );
+		$xml     = $price_method( $product );
+		// 10.00USD is the cheapest variation created by create_variation_product
+		$this->assertEquals( "<g:price>10.00USD</g:price>", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertyMpnXML() {
+		$mpn_method = $this->getProductsXmlFeedAttributeMethod( 'g:mpn' );
+		$product    = \WC_Helper_Product::create_simple_product();
+		$xml        = $mpn_method( $product );
+		$this->assertEquals( "<g:mpn>DUMMY SKU</g:mpn>", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertyAdditionalImageLinkXML() {
+		$additional_image_link_method = $this->getProductsXmlFeedAttributeMethod( 'g:additional_image_link' );
+		$product                      = \WC_Helper_Product::create_simple_product();
+
+		$xml = $additional_image_link_method( $product );
+		// By default no galery images are set.
+		$this->assertEquals( "", $xml );
+
+		// Add dummy image entry.
+		$attachment = array(
+			'post_mime_type' => 'image/png'
+		,   'post_title'     => 'product image 1'
+		);
+		$attachment_id_1 = wp_insert_attachment( $attachment, 'product_image_1.png', $product->get_id() );
+		// Product needs main image to use gallery so lets set this up here.
+		$product->set_image_id( $attachment_id_1 );
+
+		// Add second dummy image entry.
+		$attachment_id_2 = array(
+			'post_mime_type' => 'image/png'
+		,   'post_title'     => 'product image 2'
+		);
+		$attachment_id_2 = wp_insert_attachment( $attachment, 'product_image_2.png', $product->get_id() );
+
+		// Add attachment id as product image id.
+		$product->set_gallery_image_ids( [ $attachment_id_1, $attachment_id_2 ] );
+		$product->save();
+
+		$xml = $additional_image_link_method( $product );
+		$this->assertEquals( "<g:additional_image_link><![CDATA[http://example.org/wp-content/uploads/product_image_1.png,http://example.org/wp-content/uploads/product_image_2.png]]></g:additional_image_link>", $xml );
```

Comment:
We can use single quotes here (same for any of the other tests where we aren't adding any control characters or variables).